scripts/pre_commit_hook.py

```diff
@@ -67,20 +68,23 @@ def install_hook():
         python_utils.PRINT('Symlink already exists')
     else:
         try:
-            os.symlink(os.path.abspath(__file__), pre_commit_file)
+            # On Windows, it will try to copy the pyc file instead the py file.
+            this_file = __file__.replace('pyc', 'py')
+            os.symlink(os.path.abspath(this_file), pre_commit_file)
             python_utils.PRINT('Created symlink in .git/hooks directory')
         # Raises AttributeError on windows, OSError added as failsafe.
         except (OSError, AttributeError):
-            shutil.copy(__file__, pre_commit_file)
+            shutil.copy(this_file, pre_commit_file)
             python_utils.PRINT('Copied file to .git/hooks directory')
 
     python_utils.PRINT('Making pre-commit hook file executable ...')
-    _, err_chmod_cmd = start_subprocess_for_result(chmod_cmd)
+    if platform.uname()[0] != 'Windows':
```

Comment:
Why don't we want this to run on `windows` system? (Maybe we should add a comment on the top?)

Reply:
Windows system does not need these commands to make it executable.